<?xml version="1.0" encoding="UTF-8"?>
<launch>
    <!-- param -->
    <arg name="hz"                  default="10"/>
    <arg name="start_node"          default="0"/>
    <arg name="local_goal_interval" default="1.0"/>
    <arg name="local_goal_dist"     default="5.0"/>
    <!-- subscribe topic -->
    <arg name="checkpoint_topic"              default="/node_edge_map/checkpoint"/>
    <arg name="node_edge_map_topic"           default="/node_edge_map"/>
    <arg name="pose_topic"                    default="/current_pose"/>
    <arg name="reached_checkpoint_flag_topic" default="/reached_checkpoint"/>
    <arg name="local_goal_dist_topic"         default="/local_goal_dist"/>
    <!-- publish topic -->
    <arg name="local_goal_topic"         default="/local_goal"/>
    <arg name="current_checkpoint_topic" default="/current_checkpoint"/>
    <arg name="next_checkpoint_topic"    default="/next_checkpoint"/>
    <arg name="path_topic"               default="/next_checkpoint"/>

    <!-- node -->
    <node pkg="localgoal_creator" type="localgoal_creator" name="localgoal_creator" output="screen">
        <!-- param -->
        <param name="start_node"          value="$(arg start_node)"/>
        <param name="local_goal_interval" value="$(arg local_goal_interval)"/>
        <param name="local_goal_dist"     value="$(arg local_goal_dist)"/>
        <!-- topic -->
        <remap from="/checkpoint"         to="$(arg checkpoint_topic)"/>
        <remap from="/node_edge_map"      to="$(arg node_edge_map_topic)"/>
        <remap from="/current_pose"       to="$(arg pose_topic)"/>
        <remap from="/reached_checkpoint" to="$(arg reached_checkpoint_flag_topic)"/>
        <remap from="/local_goal_dist"    to="$(arg local_goal_dist_topic)"/>
        <remap from="/local_goal"         to="$(arg local_goal_topic)"/>
        <remap from="/current_checkpoint" to="$(arg current_checkpoint_topic)"/>
        <remap from="/next_checkpoint"    to="$(arg next_checkpoint_topic)"/>
        <remap from="/path"               to="$(arg path_topic)"/>
    </node>
</launch>
